joomla/joomla-cms
# [5.1] Revert implementation of AfterInitialiseRouterEvent

Pull Request for Issue #42991.

### Summary of Changes
In an ideal world, the site router would only be created when necessary with the previously added onAfterInitialiseRouter event. However if a system plugin requests the site router object before all constructors of system plugins have run, the event would be fired, but no plugin would react to it, since the events haven't been registered yet. Since we can't guarantee that no one requests the router object prematurely, we have to revert the code and load the site router regardless of the application executed.


### Testing Instructions
1. Edit `plugins/system/redirect/services/provider.php` and insert `$container->get(\Joomla\CMS\Router\SiteRouter::class);` at line 43 (after '$plugin->setDatabase($container->get(DatabaseInterface::class));') 
2. Make sure the languagefilter plugin is enabled
3. Open any page 

### Actual result BEFORE applying this Pull Request
The page throws an error.


### Expected result AFTER applying this Pull Request
The page works fine.


### Link to documentations
Please select:
